README.md

```diff
@@ -50,6 +50,23 @@ Supported installation method is via [composer](https://getcomposer.org):
 $ composer require azjezz/psl
 ```
 
+### Psalm Integration
+
+PSL comes with a [Psalm](https://psalm.dev/) plugin, that improves return type for PSL functions that psalm cannot infer from source code.
+
+To enable the Psalm plugin, add the `Psl\Internal\Psalm\Plugin` class to your psalm configuration file plugins list as follows:
+
+```xml
+<psalm>
+   ...
+   <plugins>
+      ...
+      <pluginClass class="Psl\Internal\Psalm\Plugin"/>
+   </plugins>
+</psalm>
+
+```
```

Comment:
if https://github.com/vimeo/psalm/pull/5165 gets merged, we can recommend using `vendor/bin/psalm-plugin`, as currently it is unable to discover the plugin class.

Reply:
also, we would probably wanna keep this for sometime as the change will only make it to future psalm versions.

Reply:
So for now, if you want the types to complete for the shape type, you need to manually register the plugin?
That is indeed not optimal. But I see that the PR is merged already so just a small bump on the road.

It might be better not to have to provide a plugin though ... If that were possible.